contrastive_training/contrastive_main.py

```diff
@@ -128,11 +129,12 @@
 #training loop
 #pdb.set_trace()
 if use_subsample==1:
-    cmd = "mkdir -p ./contrastive_vali_figures_"+str(training_size)+"_lmda_"+lmda_str+"_beta_"+beta_str+"_mode_"+args.mode+"_temp_"+str(args.temperature)+"_wdecay_"+str(args.weightdecay)+"_subsample"
+    cmd = "mkdir -p .\contrastive_vali_figures_"+str(training_size)+"_lmda_"+lmda_str+"_beta_"+beta_str+"_mode_"+args.mode+"_temp_"+str(args.temperature)+"_wdecay_"+str(args.weightdecay)+"_subsample"
     output_path = "./contrastive_vali_figures_"+str(training_size)+"_lmda_"+lmda_str+"_beta_"+beta_str+"_mode_"+args.mode+"_temp_"+str(args.temperature)+"_wdecay_"+str(args.weightdecay)+"_subsample/"
 else:
-    cmd = "mkdir -p ./contrastive_vali_figures_"+str(training_size)+"_lmda_"+lmda_str+"_beta_"+beta_str+"_mode_"+args.mode+"_temp_"+str(args.temperature)+"_wdecay_"+str(args.weightdecay)
+    cmd = "mkdir -p .\contrastive_vali_figures_"+str(training_size)+"_lmda_"+lmda_str+"_beta_"+beta_str+"_mode_"+args.mode+"_temp_"+str(args.temperature)+"_wdecay_"+str(args.weightdecay)
     output_path = "./contrastive_vali_figures_"+str(training_size)+"_lmda_"+lmda_str+"_beta_"+beta_str+"_mode_"+args.mode+"_temp_"+str(args.temperature)+"_wdecay_"+str(args.weightdecay)+"/"
+print(cmd)
 os.system(cmd)
```

Comment:
Don't use `os.system("mkdir blah/blah/blah")` but rather `os.makedirs("blah/blah/blah")`. More likely to be portable, less error prone and faster.

See https://docs.python.org/3/library/os.html